### PR TITLE
not dirty if auto-save soon without any save error state

### LINKED ISSUE #140069: Notebook shows dirty indicator with autosave on
- Set auto save: afterDelay
- Edit a notebook
- See the dirty indicator briefly appear after editing

### PR DESCRIPTION
**Pull request by amunger**
fixes https://github.com/microsoft/vscode/issues/140069

### CODE REVIEW

**Comment by amunger** [comment]
cc @bpasero

**Comment by bpasero** [comment]
@amunger fyi stored file working copies know if they are in error state or not, see:

https://github.com/microsoft/vscode/blob/ee5d95e9f8c969051ca2ba31637489c5489e41b3/src/vs/workbench/services/workingCopy/common/storedFileWorkingCopy.ts#L134-L140

**Comment by amunger** [comment]
> @amunger fyi stored file working copies know if they are in error state from saving or not, see:


ah, thanks. That makes things simpler. I can also include conflict state in the same way, though I never figured if that would apply to notebooks.

**Review by connor4312 — APPROVED** [review]

**Comment by bpasero** [comment]
@amunger conflict state is also an error state, so I think you are fine:

https://github.com/microsoft/vscode/blob/1e0b12d1ca02f84fd0fa836835eff8a1fb586dc2/src/vs/workbench/services/workingCopy/common/storedFileWorkingCopy.ts#L1068-L1074

File: src/vs/workbench/contrib/notebook/common/notebookEditorModel.ts [hunk 1]
```diff
@@ -133,7 +149,7 @@ export class SimpleNotebookEditorModel extends EditorModel implements INotebookE
 				this._workingCopyListeners.add(this._workingCopy.onDidChangeOrphaned(() => this._onDidChangeOrphaned.fire()));
 				this._workingCopyListeners.add(this._workingCopy.onDidChangeReadonly(() => this._onDidChangeReadonly.fire()));
 			}
-			this._workingCopy.onDidChangeDirty(() => this._onDidChangeDirty.fire(), undefined, this._workingCopyListeners);
+			this._workingCopyListeners.add(this._workingCopy.onDidChangeDirty(() => this._onDidChangeDirty.fire(), undefined, this._workingCopyListeners));
```
Comment (bpasero) [inline src/vs/workbench/contrib/notebook/common/notebookEditorModel.ts:152]: @amunger fyi this is not needed, you already pass the `this._workingCopyListeners` in

**Pull request merged**